app/Http/Controllers/RegistrationController.php

```diff
@@ -34,6 +28,31 @@ public function __construct(SettingRepository $settings)
     {
         $this->settings = $settings;
 
+        $this->middleware(['guest', 'isClosed'], ['only' => ['create']]);
+
+        $this->middleware('currentUser', ['only' => ['edit', 'update']]);
+
+        if(Auth::user())
```

Comment:
Instead of duplicating this array, it might be a bit cleaner to conditionally send `'null'` for the "user" parameter of the unique validator. That way if we have to change one of the other validation rules in the future, it only needs to be updated in one place. Something like this:

```php
$user = Auth::user();
$existingUser = isset($user) ? $user->id : 'null';
// ...
'email' => 'required|email|unique:users,email,' . $existingUser,
```